canlink/writer/writer.go

```diff
@@ -0,0 +1,68 @@
+package writer
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// Responsible for managing the trace file
+type Writer struct {
+	traceFile       *os.File
+	fileExtention   string
+	convertToString func(*zap.Logger, *TimestampedFrame) string
+	l               *zap.Logger
+}
+
+func NewWriter(l *zap.Logger, fileExtention string) *Writer {
+	convertToStringMapping := map[string]func(*zap.Logger, *TimestampedFrame) string{
+		".jsonl": convertToJson,
+		".asc":   convertToAscii,
+	}
+	return &Writer{
+		l:               l,
+		fileExtention:   fileExtention,
+		convertToString: convertToStringMapping[fileExtention],
+		traceFile:       nil,
+	}
+}
+
+func (w *Writer) CreateTraceFile(traceDir string, busName string) error {
+	file, err := createEmptyTraceFile(traceDir, busName, w.fileExtention)
```

Comment:
can you join these two lines with `w.traceFile, err := create...`?

Reply:
It was giving me errors when I tried this. err is a new variable so the := operator is valid with err, but since w.traceFile is  not a new variable, the := cant be used